File: ios/RNZendrive/Source/RNZendriveFeedback.m
//
//  RNZendriveFeedback.m
//  RNZendrive
//
//  Created by Madhusudhan Sambojhu on 02/08/19.
//

#import "React/RCTBridgeModule.h"
@interface RCT_EXTERN_MODULE(RCTZendriveFeedback, NSObject)
RCT_EXTERN_METHOD(
                  addDriveCategory: (NSString *)driveId
                  withCategory:(NSString *)category
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )
RCT_EXTERN_METHOD(
                  addEventOccurrence: (NSString *)driveId
                  withEventTimestamp:(nonnull NSNumber *)eventTimestamp
                  withEventType: (NSString *)eventType
                  withOccurrence: (BOOL *) occurrence
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject
                  )
@end

File: ios/RNZendrive/Source/RNZendriveFeedback.swift
//
//  RNZendriveFeedback.swift
//  RNZendrive
//
//  Created by Madhusudhan Sambojhu on 02/08/19.
//

import Foundation
import ZendriveSDK
@objc(RCTZendriveFeedback)
class RNZendriveFeedback: NSObject {

    @objc
    func constantsToExport() -> [AnyHashable: Any]! {
        return [:]
    }

    @objc
    func addDriveCategory(
        _ driveId: String,
        withCategory category: String,
        resolver resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: RCTPromiseRejectBlock
    ) {
        ZendriveFeedback.addDriveCategory(withDriveId: driveId, driveCategory: Utils.feedbackCategory(of: category))
        resolve(true)
    }

    // swiftlint:disable function_parameter_count
    @objc
    func addEventOccurrence(
        _ driveId: String,
        withEventTimestamp eventTimestamp: NSNumber,
        withEventType eventType: String,
        withOccurrence occurrence: Bool,
        resolver resolve: @escaping RCTPromiseResolveBlock,
        rejecter reject: RCTPromiseRejectBlock
    ) {
        ZendriveFeedback.addEventOccurrence(withDriveId: driveId, eventTimestamp: Int64(truncating: eventTimestamp),
                                            eventType: Utils.eventType(of: eventType), occurrence: occurrence)
        resolve(true)
    }
    // swiftlint:enable function_parameter_count

    @objc
    static func requiresMainQueueSetup() -> Bool {
        return false
    }
}
